tests/foreman/api/test_contentmanagement.py

```diff
@@ -148,6 +148,32 @@ def update_capsule_download_policy(self, capsule_id, download_policy):
         proxy.download_policy = download_policy
         proxy.update(['download_policy'])
 
+    @tier4
+    def test_positive_insights_puppet_package_availability(self):
+        """Check `redhat-access-insights-puppet` package availability for
+        capsule
+
+        :BZ: 1315844
+
+        :id: a31b0e21-aa5d-44e2-a408-5e01b79db3a1
+
+        :expectedresults: `redhat-access-insights-puppet` package is delivered
+            in capsule repo and is available for installation on capsule via
+            yum
+
+        :CaseLevel: System
+        """
+        package_name = 'redhat-access-insights-puppet'
+        result = ssh.command(
```

Comment:
why not:
```python
result = self.capsule_vm.run('yum list available | grep {}'.format(package_name))
```

Reply:
Any way to re-init existing capsule? 🙂 
I don't want to wait for another hour between every single test run, `ssh.command` allows me to comment capsule initialization for debugging purposes.

Reply:
are you finish with PR?

Reply:
Yep, still, all the tests in this module follow the same style, it'll be a nightmare to debug them otherwise. And even setup class contains instructions on how to reuse existing capsule easily, if we introduce `self.capsule_vm.run()` commands such reuse won't be possible so i'm strongly against using `capsule_vm.run()` here.
Btw i'd rather like to see a refactor of capsule module so we could re-init existing one in the same manner as nailgun entities (smth like `self.capsule = Capsule(id=2, name='example.com')`), that'd unblock us here.

Reply:
can you briefly explain why it is problem to use `self.capsule_vm.run()`? I can see in setup `cls.capsule_vm.create()`, so it is already created and all time necessary already spent

Reply:
We usually have few hosts with capsule installed dedicated for testing. If some tests are failing on jenkins, or you need to debug something - you can simply pass hostname in setupclass and start executing the tests in a matter of seconds.
If we start using `self.capsule_vm.run` - we'll remove the ability to re-use any existing capsule at all and will be forced to create a new one every single time we want to execute the test.
Test is failing on jenkins and you want to reproduce that failure locally? Cool, wait for 1 hour. Reproduced and now want to use some breakpoints for debugging? Wait another hour for capsule creation. Made a fix but there was a typo and test failed? Another hour. Wanna grab test results for your PR? One more hour.
`run()` command will create us terrible limitations with no benefits at all, and basically `run()` is actually a wrapper for `ssh.command()` which just passes the hostname.

tl;dr - using `vm.run()` will make entire module so enjoyable to work with, as hammer's `test_positive_all_options` was, with no benefits introduced at all.

Reply:
that still makes no sense to me
I have created new capsule using `vm.create()` command and now I want to run something on it. `vm.run` should take few seconds to execute, so not sure about logic mentioned

Reply:
vm.run == ssh.command(hostname=vm.hostname), the difference is vm.run can _only_ be used in case the vm you're trying to point to was created by `vm.create` in the same test run previously; you can't pass any existing vm to save you hour of time.

Reply:
and why vm from setupclass is not in the same test run?